polaris/dataset/_factory.py

```diff
@@ -0,0 +1,229 @@
+import os
+from typing import Dict, Optional
+
+import datamol as dm
+import pandas as pd
+import zarr
+from loguru import logger
+
+from polaris.dataset import ColumnAnnotation, Dataset
+from polaris.dataset._adapters import Adapter
+from polaris.dataset.converters import Converter, SDFConverter, ZarrConverter
+
+
+def create_dataset_from_file(path: str, zarr_root_path: Optional[str] = None) -> Dataset:
+    """
+    This function is a convenience function to create a dataset from a file.
+
+    It sets up the dataset factory with sensible defaults for the converters.
+    For creating more complicated datasets, please use the `DatasetFactory` directly.
+    """
+    factory = DatasetFactory(zarr_root_path=zarr_root_path)
+    factory.register_converter("sdf", SDFConverter())
+    factory.register_converter("zarr", ZarrConverter())
+
+    factory.add_from_file(path)
+    return factory.build()
+
+
+class DatasetFactory:
+    """
+    The `DatasetFactory` makes it easier to create complex datasets.
+
+    It is based on the the factory design pattern and allows a user to specify specific handlers
+    (i.e. [`Converter`][polaris.dataset.converters._base.Converter] objects) for different file types.
+    These converters are used to convert commonly used file types in drug discovery
+    to something that can be used within Polaris while losing as little information as possible.
+
+    In addition, it contains utility method to incrementally build out a dataset from different sources.
+
+    Tip: Try quickly converting one of your datasets
+        The `DatasetFactory` is designed to give you full control.
+        If your dataset is saved in a single file and you don't need anything fancy, you can try use
+        [`create_dataset_from_file`][polaris.dataset.create_dataset_from_file] instead.
+
+        ```py
+        from polaris.dataset import create_dataset_from_file
+        dataset = create_dataset_from_file("path/to/my_dataset.sdf")
+        ```
+
+    Question: How to make adding meta-data easier?
+        The `DatasetFactory` is designed to more easily pull together data from different sources.
+        However, adding meta-data remains a laborous process. How could we make this simpler through
+        the Python API?
+    """
+
+    def __init__(self, zarr_root_path: Optional[str] = None) -> None:
```

Comment:
**Suggestion**
Perhaps it would make this API more flexible if the user could specify the converters they want directly here, when initializing the factory?